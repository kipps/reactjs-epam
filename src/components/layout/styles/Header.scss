.Header {
  margin-bottom: 24px;
  padding: 16px;
  background: #A770EF;  /* fallback for old browsers */
  background: -webkit-linear-gradient(to right, #FDB99B, #CF8BF3, #A770EF);  /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(to right, #FDB99B, #CF8BF3, #A770EF); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
}

input {
  width: 100%;
  display: block;
  padding: 12px 24px;
  font-size: 1em;
  margin-bottom: 16px;
}


body {
  margin: 0;
  padding: 0;
}
