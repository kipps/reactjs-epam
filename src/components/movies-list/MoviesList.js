import React from 'react';
import {connect} from "react-redux";
import PropTypes from "prop-types";
import store from '../../redux/store/configureStore'
import {fetchPostsRequest, searchByTitle, headerSearchSet} from "../../redux/actions/MoviesAction";
import Movie from '../movie/Movie';
import Container from 'react-bootstrap/Container';
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";


class MoviesList extends React.Component {
    componentDidMount() {
        let path = window.location.pathname;
        path.includes('/search/') ? store.dispatch(searchByTitle(path.replace('/search/', ''))) : store.dispatch(fetchPostsRequest());
        store.dispatch(headerSearchSet(true))
    }

    componentDidUpdate(prevProps) {
        if (this.props.location.pathname !== prevProps.location.pathname) {
            console.log('route changed');
            let path = window.location.pathname;
            path.includes('/search/') ? store.dispatch(searchByTitle(path.replace('/search/', ''))) : store.dispatch(fetchPostsRequest());
            store.dispatch(headerSearchSet(true));
        }
    }

    render() {
        return (this.props.loading) ?
            <Container><h1 className='flex f-large v-center h-center'>Loading Country Names...</h1></Container> :
            (!this.props.posts.length) ?
                <Container><h1 className='f-large'> No Movies </h1></Container> :
                <Container>
                    <Row>
                        {this.props.posts.map(
                            item => (
                                <Col key={item.id} md={3}>
                                    <Movie path={'/film/'} {...item} />
                                </Col>
                            )
                        )}
                    </Row>
                </Container>
    }
}

MoviesList.propTypes = {
    posts: PropTypes.array,
    loading: PropTypes.bool,
    headerSearchShow: PropTypes.bool
}

const mapStateToProps = state => {
    return {
        posts: state.moviesState.posts,
        loading: state.moviesState.loading,
        headerSearchShow: state.moviesState.headerSearchShow
    }
}

export default connect(mapStateToProps)(MoviesList)