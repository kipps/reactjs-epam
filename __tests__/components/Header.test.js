import {mount, shallow} from 'enzyme/build';
import { configure } from 'enzyme/build';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16/build';
import Header from "../../src/components/layout/header";
import MoviesList from "../../src/components/movies-list/MoviesList";

configure({ adapter: new Adapter() });

jest.useFakeTimers();

describe("Header ", () => {
  let component = mount(<Header />);

  it("renders default header", () =>
    expect(
      shallow(<Header />)
        .find('header.Header')
        .length
    ).toBe(1));

  it('renders header correctly', () => {
    const tree = renderer
      .create(<Header />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});