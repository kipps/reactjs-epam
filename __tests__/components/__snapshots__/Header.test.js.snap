// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header  renders header correctly 1`] = `
<header
  className="Header pt-24 pb-24"
>
  <div
    className="container"
  >
    <div
      className="mb-24 row"
    >
      <div
        className="flex v-center col"
      >
        <h1
          className="white f-large"
        >
          Movies for fun
        </h1>
      </div>
      <div
        className="flex v-center h-right col"
      >
        <div
          className="User"
        >
          <a
            className="User__link black"
            href="#"
          >
            Roman Shevchenko
          </a>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <input
          className="full"
          placeholder="Finde your movie"
          type="search"
        />
        <div
          className="flex space-between v-center pt-8"
        >
          <div
            className="flex v-center"
          >
            <span>
              Search by:
            </span>
            <div
              className="ml-8 Radio"
            >
              <label
                className="inline ml-8 mr-8"
                name="sort"
              >
                <span
                  className="inline mr-4"
                >
                  Genre
                </span>
                <input
                  name="sort"
                  type="radio"
                />
              </label>
              <label
                className="inline ml-8 mr-8"
                name="sort"
              >
                <span
                  className="inline mr-4"
                >
                  Title
                </span>
                <input
                  name="sort"
                  type="radio"
                />
              </label>
              <label
                className="inline ml-8 mr-8"
                name="sort"
              >
                <span
                  className="inline mr-4"
                >
                  Other
                </span>
                <input
                  name="sort"
                  type="radio"
                />
              </label>
            </div>
          </div>
          <div>
            <button
              className="btn btn-danger btn-sm"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Search
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
